backend/src/main/java/com/redhat/cloud/notifications/db/builder/JoinBuilder.java

```diff
@@ -0,0 +1,70 @@
+package com.redhat.cloud.notifications.db.builder;
+
+public class JoinBuilder {
+
+    private final StringBuilder query = new StringBuilder();
+
+    JoinBuilder() {
+
+    }
+
+    public static JoinBuilder builder() {
+        return new JoinBuilder();
+    }
+
+    public JoinBuilder leftJoin(String what) {
+        addSpaceIfNeeded();
+        query.append("LEFT JOIN ").append(what);
+
+        return this;
+    }
+
+    public JoinBuilder join(String what) {
+        addSpaceIfNeeded();
+        query.append("JOIN ").append(what);
+
+        return this;
+    }
+
+    public JoinBuilder ifLeftJoin(boolean condition, String what) {
+        if (condition) {
+            leftJoin(what);
+        }
+
+        return this;
+    }
+
+    public JoinBuilder ifJoin(boolean condition, String what) {
```

Comment:
I wanted to mimic the regular flow:
```
if (needsJoin) {
   query += " JOIN stuff";
}
```

=>  `ifJoin(needsJoin, "stuff")`

but open to suggestons, is there any reason you think the later is better?

Reply:
`ifJoin` feels a bit less clear to me than `joinIf` because `ifJoin` could be misinterpreted as `if there's a join, then...` at first reading for a new API user. But that's really not important at all, I can live with `ifJoin` 😄